FirebaseDatabase/Sources/Utilities/FUtilities.m

```diff
@@ -154,9 +154,13 @@ + (FParsedUrl *)parseUrl:(NSString *)url {
 
     // Sanitize the database URL by removing the path component, which may
     // contain invalid URL characters.
+    NSRange lastMatch = [url rangeOfString:originalPathString
+                                   options:NSBackwardsSearch];
     NSString *sanitizedUrlWithoutPath =
-        [url stringByReplacingOccurrencesOfString:originalPathString
-                                       withString:@""];
+        (lastMatch.location != NSNotFound)
+            ? [url stringByReplacingCharactersInRange:lastMatch withString:@""]
```

Comment:
Isn't this just a substring now? I think it might be easier to parse if we use `substring(with: NSRange)`.

Reply:
Good catch. Thanks!